tools/pd-ctl/pdctl/command/keyspace_command.go

```diff
@@ -15,33 +15,216 @@
 func NewKeyspaceCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "keyspace [command] [flags]",
-		Short: "show keyspace information",
-		Run:   showKeyspaceCommandFunc,
+		Use:   "keyspace <command> [flags]",
+		Short: "keyspace commands",
 	}
+	cmd.AddCommand(newShowKeyspaceCommand())
+	cmd.AddCommand(newCreateKeyspaceCommand())
+	cmd.AddCommand(newUpdateKeyspaceConfigCommand())
+	cmd.AddCommand(newUpdateKeyspaceStateCommand())
+	cmd.AddCommand(newListKeyspaceCommand())
 	return cmd
 }
 
+func newShowKeyspaceCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "show [flags]",
+		Short: "show keyspace metadata specified by keyspace name/id, if both provided, name will be used",
+		Run:   showKeyspaceCommandFunc,
+	}
+	r.Flags().String(nmID, "", "keyspace id")
+	r.Flags().String(nmName, "", "keyspace name")
+	return r
+}
+
 func showKeyspaceCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 0 {
+		cmd.Usage()
+		return
+	}
+	id, err := cmd.Flags().GetString(nmID)
+	if err != nil {
+		cmd.Println("Failed to parse flag: ", err)
+		return
+	}
+	name, err := cmd.Flags().GetString(nmName)
+	if err != nil {
+		cmd.Println("Failed to parse flag: ", err)
+		return
+	}
+	var url string
+	if name != "" {
+		url = fmt.Sprintf("%s/%s?force_refresh_group_id=true", keyspacePrefix, name)
+	} else if id != "" {
+		url = fmt.Sprintf("%s/id/%s", keyspacePrefix, id)
+	} else {
+		cmd.Println("Either id or name should be provided")
+		return
+	}
+	resp, err := doRequest(cmd, url, http.MethodGet, http.Header{})
+	if err != nil {
+		cmd.Printf("Failed to get the keyspace information: %s\n", err)
+		return
+	}
+	cmd.Println(resp)
+}
+
+func newCreateKeyspaceCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "create <keyspace-name> [flags]",
+		Short: "create a keyspace",
+		Run:   createKeyspaceCommandFunc,
+	}
+	r.Flags().String(nmConfig, "", "keyspace config, in json format")
+	return r
+}
+
+func createKeyspaceCommandFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 1 {
 		cmd.Usage()
 		return
 	}
+	configStr, err := cmd.Flags().GetString(nmConfig)
+	if err != nil {
+		cmd.Println("Failed to parse flag: ", err)
+		return
+	}
+	var config map[string]string
+	if err = json.Unmarshal([]byte(configStr), &config); err != nil {
+		cmd.Println("Failed to parse flag: ", err)
+		return
+	}
+	params := handlers.CreateKeyspaceParams{
+		Name:   args[0],
+		Config: config,
+	}
+	body, err := json.Marshal(params)
+	if err != nil {
+		cmd.Println(err)
+		return
+	}
+	resp, err := doRequest(cmd, keyspacePrefix, http.MethodPost, http.Header{}, WithBody(bytes.NewBuffer(body)))
+	if err != nil {
+		cmd.Printf("Failed to create the keyspace: %s\n", err)
+		return
+	}
+	cmd.Println(resp)
+}
+
+func newUpdateKeyspaceConfigCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "update-config <keyspace-name> <config-patch>",
+		Short: "update keyspace config, a json merge patch is expected",
```

Comment:
Is it more convenient to use JSON or use key-value list? BTW, do we need a force overwrite option?

Reply:
I have changed the json to comma separated kv pairs.